### ISSUE TITLE
nixos users.users.<name>.packages doesn't work with symlinks

### ISSUE THREAD

**Issue by Gerschtli** [NixOS/nixpkgs#32728]
## Issue description

Libreoffice and sublime text 3 have symlinks in their out path. The problem is that these symlinks don't get dereferenced. Therefore the files and folders behind this symlink won't be linked and after every `nixos-rebuild` I get
```
/nix/store/9vnczgbam4g81j9ngcah2icv4d9mx5nk-libreoffice-5.2.6.2/share:
applications: File exists
/nix/store/gflz6ra61y0ymzrs65kfgz238bhamk6v-sublimetext3-3143/share:
icons: File exists
```

### Steps to reproduce

I installed these packages as user packages:
```nix
{ pkgs, ...}:
{
  users.users.tobias.packages = with pkgs; [ libreoffice sublime3 ];
}
```

More information:
```
$> ll /nix/store/9vnczgbam4g81j9ngcah2icv4d9mx5nk-libreoffice-5.2.6.2/share
total 16K
lrwxrwxrwx 2 root root   89 Jan  1  1970 applications -> /nix/store/9vnczgbam4g81j9ngcah2icv4d9mx5nk-libreoffice-5.2.6.2/lib/libreoffice/share/xdg/
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 desktop/
dr-xr-xr-x 3 root root 4.0K Jan  1  1970 gsettings-schemas/
dr-xr-xr-x 4 root root 4.0K Jan  1  1970 icons/

$> ll /nix/store/9vnczgbam4g81j9ngcah2icv4d9mx5nk-libreoffice-5.2.6.2/share/applications/
total 48K
-r--r--r-- 4 root root 3.7K Jan  1  1970 base.desktop
-r--r--r-- 4 root root 4.7K Jan  1  1970 calc.desktop
-r--r--r-- 4 root root 3.7K Jan  1  1970 draw.desktop
-r--r--r-- 4 root root 4.3K Jan  1  1970 impress.desktop
-r--r--r-- 4 root root 3.3K Jan  1  1970 math.desktop
-r--r--r-- 4 root root 1.1K Jan  1  1970 qstart.desktop
-r--r--r-- 4 root root 3.9K Jan  1  1970 startcenter.desktop
-r--r--r-- 4 root root 4.9K Jan  1  1970 writer.desktop
-r--r--r-- 4 root root  928 Jan  1  1970 xsltfilter.desktop

$> ll /etc/per-user-pkgs/tobias/share/applications/
total 44K
lrwxrwxrwx 1 root root  94 Jan  1  1970 Eclipse.desktop -> /nix/store/mwgrx51j59xp03ckavnp8lapyzcyjjmx-eclipse-sdk-4.7/share/applications/Eclipse.desktop
lrwxrwxrwx 1 root root  87 Jan  1  1970 atom.desktop -> /nix/store/kh99mazsn31b0gkjdl99dxgcx5c5x11f-atom-1.23.0/share/applications/atom.desktop
lrwxrwxrwx 1 root root 101 Jan  1  1970 dropbox.desktop -> /nix/store/8bb1z2111dg4pbc9552l1lxgz1rcnbzv-dropbox-cli-2015.10.28/share/applications/dropbox.desktop
lrwxrwxrwx 1 root root  87 Jan  1  1970 gimp.desktop -> /nix/store/xvnnidwq71i0s68wdlb8m6zipm17wajn-gimp-2.8.22/share/applications/gimp.desktop
lrwxrwxrwx 1 root root 111 Jan  1  1970 google-chrome.desktop -> /nix/store/dpaj0ib5x06nvzf966csvk22qm7mcpgi-google-chrome-63.0.3239.84/share/applications/google-chrome.desktop
lrwxrwxrwx 1 root root  98 Jan  1  1970 pavucontrol.desktop -> /nix/store/k8jwvgh35lbm7jraw5wyfy1wkwxgwn9c-pavucontrol-3.0/share/applications/pavucontrol.desktop
lrwxrwxrwx 1 root root  95 Jan  1  1970 qpdfview.desktop -> /nix/store/kjngp65z5987ayab9kv8c15b5d0ii2w6-qpdfview-0.4.16/share/applications/qpdfview.desktop
lrwxrwxrwx 1 root root 110 Jan  1  1970 spotify.desktop -> /nix/store/0fkd4dl6vpxzcxzhgbby0jylm502gcim-spotify-1.0.64.407.g9bd02c2d-26/share/applications/spotify.desktop
lrwxrwxrwx 1 root root  86 Jan  1  1970 sublime_text.desktop -> /nix/store/ba90x6yc9az3ry7cf5n1c8r1cp3z35br-sublimetext3-3143-bin/sublime_text.desktop
lrwxrwxrwx 1 root root 101 Jan  1  1970 thunderbird.desktop -> /nix/store/jbfxspb1kkc3nghzk68lfm2hlkvk3wpa-thunderbird-52.5.0/share/applications/thunderbird.desktop
lrwxrwxrwx 1 root root  86 Jan  1  1970 xterm.desktop -> /nix/store/xyg0916kgvvi65xrk0fk8j1mznkyj4mk-xterm-330/share/applications/xterm.desktop*
```

The same goes for sublime.
```
$> ll /nix/store/gflz6ra61y0ymzrs65kfgz238bhamk6v-sublimetext3-3143/share                                                                                                 
total 8.0K                                                                                                                                                              
dr-xr-xr-x 2 root root 4.0K Jan  1  1970 applications/                                                                                                                  
lrwxrwxrwx 1 root root   79 Jan  1  1970 icons -> /nix/store/ba90x6yc9az3ry7cf5n1c8r1cp3z35br-sublimetext3-3143-bin/Icon/256x256//                                    

$> ll /nix/store/gflz6ra61y0ymzrs65kfgz238bhamk6v-sublimetext3-3143/share/icons/                                                                              
total 32K                                                                                                                                                     
-r--r--r-- 4 root root 31K Jan  1  1970 sublime-text.png                           
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.68, NixOS, 17.09.2378.af7e47921c4 (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.15`
 - channels(root): `"nixos-17.09.2378.af7e47921c4"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by vcunat** [comment]
This is a duplicate of https://github.com/NixOS/nixpkgs/issues/31253 which has been fixed on nixos-unstable but not backported to 17.09 (yet).

**Comment by Gerschtli** [comment]
Closing this issue because fix is available in 18.03
